src/cpp/rtps/RTPSDomain.cpp

```diff
@@ -167,6 +168,15 @@ RTPSParticipant* RTPSDomain::createParticipant(
     // Generate a new GuidPrefix_t
     GuidPrefix_t guidP;
     guid_prefix_create(ID, guidP);
+    if (!PParam.builtin.metatraffic_external_unicast_locators.empty())
```

Comment:
I understand this is to prevent two identical containers in different hosts to try and use SHM, but at the same time, doesn't it prevent the use of SHM between two participants in the same container if at least one of them configures external locators?

Reply:
Yes. It prevents the use of SHM between participants in the same "host" that have different addresses on their metatraffic external locators.

There is a compromise here. We could remove commits 13fb706b9a8ce47384f994dbbdb931ecd686e150 and c46658ff3c84e3eed5a7a87ce0ab84910cc2bd5a, but in that case I think two identical containers in different hosts might try to talk using intraprocess ...

Reply:
I think we'll need to leave it as it is. Please consider adding a note in the documentation